torch/csrc/jit/passes/onnx/shape_type_inference.cpp

```diff
@@ -703,15 +703,21 @@ std::vector<::c10::ShapeSymbol> Broadcast(
     if (is_static_0 && is_static_1) {
       int64_t static_0_sz = ss_shape_0.static_size();
       int64_t static_1_sz = ss_shape_1.static_size();
-      final_shape[rank_max - 1 - idx] = ::c10::ShapeSymbol::fromStaticSize(
-          std::max(static_0_sz, static_1_sz));
+      // condition for corner case of 0d tensor
+      // 0d tensor with 1d tensor would give us 0d tensor
+      if (std::min(static_0_sz, static_1_sz) == 0) {
+        final_shape[rank_max - 1 - idx] = ::c10::ShapeSymbol::fromStaticSize(
```

Comment:
nit: could we use shape_idx globally to represent `rank_max - 1 - idx` for simplicity?
```
size_t shape_idx = rank_max - 1 - idx;
```

Reply:
Done